lib/ecto/migration.ex

```diff
@@ -828,6 +843,7 @@ defmodule Ecto.Migration do
       raise ArgumentError, "column #{Atom.to_string(column)} is missing precision option"
     end
 
+    validate_type!(type)
```

Comment:
The same type validations relevant to column additions would also seem to be relevant to column modifications.